File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        result = True

        # So what we did here is we checked the length from left side and right side and then compare it and if it is greater than one that means that the difference between 2 sides of the binary tree is greater than 1 that means it is not a balanced tree so we checked height from either of the side and and checked the difference between them and if it is greater than one we return false otherwise true here we added nonlocal keyword that means it can use the outside result as its local inside the function.

        # How the height is checked is? If root is null then return 0 and increment it on every previous recursion 

        def height(root):
            nonlocal result

            if root == None:
                return 0

            l = height(root.left)
            r = height(root.right)

            if abs(l-r) > 1:
                result = False
            
            return max(l,r)+1
        
        height(root)
        return result

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    var result = true

    var checkRoot = root => {
        if(root == null){
            return 0
        }

        let l = checkRoot(root.left)
        let r = checkRoot(root.right)

        if(Math.abs(l-r) > 1){
            result = false
        }

        return Math.max(l,r) + 1
    }

    checkRoot(root)
    return result 
};